btx/diagnostics/run.py

```diff
@@ -429,6 +454,77 @@ def visualize_energy_stats(self, output=None, outlier_threshold=-np.inf):
             if output is not None:
                 fig.savefig(output, dpi=300, bbox_inches='tight')
     
+    def visualize_stats_interactive(self, y_key='mean', color_by='beam_energy_eV'):
+        """
+        Plot trajectories of run statistics, interactively.
+
+        Parameters
+        ----------
+        y_key : str
+            self.stats_final key to plot
+        color_by : str
+            self.stats_final key used to color
+        """
+        hv.extension('bokeh')
+
+        y = self.stats_final[y_key]
+
+        df = pd.DataFrame({
+            'evt_id' : np.arange(y.shape[0]), 
+            'y' : self.stats_final[y_key], 
+            'color_by' : self.stats_final[color_by]})
+        scatter = hv.Scatter(df, kdims=['evt_id'], vdims=['y', 'color_by'])
+        return scatter.opts(opts.Scatter(tools=['hover'])).opts(width=800, color_index='color_by')
+
+    def display_img_evt(self, event_id, 
+            vmin=1, vmax=1e3, figsize=8, dpi=360, title=None, log=True, mask_negatives=True,
+            method='pyplot'):
+        """
+        Display the detector image corresponding to a given event ID.
+
+        Parameters
+        ----------
+        event_id : int
+            Index of the desired event
+        vmin : float
+            Lowest pixel value
+        figsize : int
+            Figure size
+        dpi : int
+            Figure dots per inch
+        title : str or None
+            Figure title
+        log : bool
+            Whether the detector image is displayed on linear or log scale.
+        mask_negatives: bool
+            Whether negative intensity pixels should be masked.
+        method: str
+            options: 'pyplot' (default) or 'holoviews'
+        """
+        evt = self.psi.runner.event(self.psi.times[event_id])
+        img = self.psi.det.image(evt=evt)
+        if mask_negatives:
+            img = np.ma.masked_where(img<=0,img)
+
+        if method is 'pyplot':
```

Comment:
It might be better to use `==` rather than `is` here since you're testing for equality (and not against `None`)